guides/doc-Administering_Red_Hat_Satellite/topics/Renaming_a_Satellite_Server.adoc

```diff
@@ -10,7 +10,7 @@ For more information about configuring external authentication, see xref:Configu
 
 If you use virt-who, you must update the virt-who configuration files with the new host name after you run the `{project-change-hostname}` script.
 ifndef::orcharhino[]
-For more information, see https://access.redhat.com/documentation/en-us/red_hat_satellite/{AccessRedHatComVersion}/html/configuring_virtual_machine_subscriptions_in_red_hat_satellite/troubleshooting-virt-who#modifying-virt-who-configuration_vm-subs-satellite[Modifying a virt-who Configuration] in _Configuring Virtual Machine Subscriptions in {ProjectName}_.
+For more information, see {MultiBaseURL}/configuring_virtual_machine_subscriptions_in_red_hat_satellite/troubleshooting-virt-who#modifying-virt-who-configuration_vm-subs-satellite[Modifying a virt-who Configuration] in _Configuring Virtual Machine Subscriptions in {ProjectName}_.
```

Comment:
In Foreman/Katello context, this does get rendered as a link to RH Portal, is this intended? I would expect linking the upstream guide instead.

In other words, I think the best solution would be also to override `MultiBaseURL` in non-Satellite context with `BaseURL`. Upstream there is no multi and single-page we just have one version for all.

Reply:
> In Foreman/Katello context, this does get rendered as a link to RH Portal, is this intended? I would expect linking the upstream guide instead.

I would expect the same thing, but note that it was already there prior to this PR. I don't have a good solution to this now but you're completely right that we should link to an upstream document.

Looking at the documentation, we may be able to use a custom macro. https://dzone.com/articles/asciidoctor-creating-a-macro describes how it works with asciidoctor and https://asciidoc-py.github.io/userguide.html#_macros has more. However, this may not work with the other systems (AFAIK Orcharhino uses Antora and RH uses something else).

> In other words, I think the best solution would be also to override MultiBaseURL in non-Satellite context with BaseURL. Upstream there is no multi and single-page we just have one version for all.

I agree, but I'm not sure how to technically achieve that now.

Reply:
My thoughts: Let's not overcomplicate it for now with a macro if we can use an attribute instead. On the other hand, if it's only for links to access.redhat.com, I am fine because those should only be visible in Satellite builds anyway.

But yes, please keep the downstream solutions Antora and pantheon in mind.

Reply:
AFAIK most guides have upstream equivalents. We just don't build the multi guide. For example:

```
guides/doc-Configuring_Load_Balancer/topics/Before_You_Begin.adoc
29:To upgrade {SmartProxyServer}s from {ProductVersionPrevious} to {ProductVersion}, complete the {MultiBaseURL}/upgrading_and_updating_red_hat_satellite/upgrading_red_hat_satellite#upgrading_capsule_server[Upgrading {SmartProxyServer}s] procedure in _Upgrading and Updating {ProjectName}_.
```

This refers to what we build from `guides/doc-Upgrading_and_Updating` but then a specific chapter of it. So essentially what you have is:
```asciidoc
ifdef::satellite[]
To upgrade {SmartProxyServer}s from {ProductVersionPrevious} to {ProductVersion}, complete the {MultiBaseURL}/upgrading_and_updating_red_hat_satellite/upgrading_red_hat_satellite#upgrading_capsule_server[Upgrading {SmartProxyServer}s] procedure in _Upgrading and Updating {ProjectName}_.
endif::[]
ifndef::satellite[]
To upgrade {SmartProxyServer}s from {ProductVersionPrevious} to {ProductVersion}, complete the {UpgradingDocURL}#upgrading_capsule_server[Upgrading {SmartProxyServer}s] procedure in _Upgrading and Updating {ProjectName}_.
endif::[]
```

Or in short: `{MultiBaseURL}/upgrading_and_updating_red_hat_satellite/upgrading_red_hat_satellite#upgrading_capsule_server` vs `{UpgradingDocURL}#upgrading_capsule_server`.

Then when you take in mind that `UpgradingDocURL` is defined as:
```
guides/common/attributes-base.adoc
14::UpgradingDocURL: {BaseURL}Upgrading_and_Updating/{BaseFilenameURL}#

guides/common/attributes-satellite.adoc
12::UpgradingDocURL: {BaseURL}upgrading_and_updating_red_hat_satellite/index#
```

You see that it's even a smaller difference. It's really the base URL and `upgrading_red_hat_satellite` vs `{BaseFilenameURL}` (you could say `index` is `{BaseFilenameURL}` for satellite).

Reply:
I thought this PR was purely about links to docs that are no (yet) part of upstream.

Of course we should keep our attributes the way they are; i.e. one attribute per guide. You can decide to use single vs multi per build, e.g. in Satellite attributes.

The way it is right now if IMHO great because it's short and you only need to know which guide it is in.

Reply:
> I thought this PR was purely about links to docs that are no (yet) part of upstream.

In the beginning I thought so too but over time I found out the real issues. For example, 3daa833e3381c95517de9594d634c9f0d26b4b93 shows that things never linked to the upstream guide that was already there. Then there are still a bunch of links it doesn't replace since I don't know how to deal with multi vs single.

For example, `guides/doc-Planning_Guide/topics/Capsule_Server_Overview.adoc` has:
```asciidoc
ifdef::satellite[]
The {InstallingProjectDocURL}satellite-ports-and-firewalls-requirements_{project-context}[Ports and Firewalls Requirements] in _{project-installation-guide-title}_, https://access.redhat.com/documentation/en-us/red_hat_satellite/{AccessRedHatComVersion}/html/installing_satellite_server_from_a_disconnected_network/preparing-environment-for-satellite-installation#satellite-ports-and-firewalls-requirements_{project-context}[Ports and Firewalls Requirements] in _Installing {ProjectServer} from a Disconnected Network_ and {InstallingSmartProxyDocURL}capsule-ports-and-firewalls-requirements_{smart-proxy-context}[Ports and Firewalls Requirements] in _{smart-proxy-installation-guide-title}_ contain complete instructions for configuring the host-based firewall to open the ports required.

A matrix table of ports is available in the Red{nbsp}Hat Knowledgebase solution https://access.redhat.com/solutions/5627751[Red Hat Satellite List of Network Ports].
endif::[]
```

IMHO this is wrong since it's not Satellite specific and all guides that it mentions are upstream. The matrix of ports was actually introduced upstream in #761 and the link to the knowledge base that remains is probably an oversight.

However, we don't have an attribute that can replace `https://access.redhat.com/documentation/en-us/red_hat_satellite/{AccessRedHatComVersion}/html/installing_satellite_server_from_a_disconnected_network/preparing-environment-for-satellite-installation#satellite-ports-and-firewalls-requirements_{project-context}`. You can end up with an ugly solution such as https://github.com/theforeman/foreman-documentation/pull/924 but this may not work for sentences that can't be split up.

What really needs to be done is to create a tracker to further eliminate all links to downstream documentation in upstream and figure out a solution. However, I won't have time to work on that.